doc/asyncomplete.txt

```diff
@@ -14,80 +14,76 @@ CONTENTS                                                *asyncomplete-contents*
-
-    Type: |Number|
-    Default: `0`
-
-    Remove duplicates from the completion list keeping the match from the
-    source with the highest priority. Set to `1` to enable.
-
-    Note that this might be slow if you have large and/or many sources!
-
-
 g:asyncomplete_auto_popup                           *g:asyncomplete_auto_popup*
 
     Type: |Number|
     Default: `1`
 
     Automatically show the autocomplete popup menu as you start typing.
 
-g:asyncomplete_default_refresh_pattern  *g:asyncomplete_default_refresh_pattern*
+g:asyncomplete_log_file                               *g:asyncomplete_log_file*
 
     Type: |String|
-    Default: `'\(\k\+$\|\.$\|:$\)'`
+    Default: null
 
-    Default refresh pattern to use for when to show the popup menu.
+    Path to log file.
 
-g:asyncomplete_smart_completion                  *g:asyncomplete_smart_completion*
+g:asyncomplete_popup_delay                         *g:asyncomplete_popup_delay*
 
     Type: |Number|
-    Default: 0
+    Default: 30
 
-    Enable smart completion i.e. fuzzy match.
-    Requirements:
-        * Vim8 with lua support `:echo has('lua')` or Neovim 0.2.2+
-        * TextChangedP autocmd support - `:echo exists('##TextChangedP')`
-        * `let g:asyncomplete_auto_popup = 1`
+    Milliseconds to wait before opening the popup menu
 
-    Feel free to suggest or send pull requests to improve the smart completion
-    algorithm.
-
-g:asyncomplete_force_refresh_on_context_changed *g:asyncomplete_force_refresh_on_context_changed*
+g:asyncomplete_auto_completeopt               *g:asyncomplete_auto_completeopt*
 
     Type: |Number|
-    Default: `0`
+    Default: 1
 
-    Automatically force refresh the autocomplete if the context has
-    already changed when results were received.
+    Set default `completeopt` options. These are `menuone,noinsert,noselect`.
+    This effectively overwrites what ever the user has in their config file.
 
-    Note Setting to 1 may call force_refresh often which may slow down vim
-    specially for those sources that are heavy in computation in vimscript.
+    Set to 0 to disable.
 
-g:asyncomplete_log_file                               *g:asyncomplete_log_file*
+g:asyncomplete_preprocessor                        *g:asyncomplete_preprocessor*
 
-    Type: |String|
-    Default: null
+    Type: |Array| for zero or one |Function|
+    Default: []
 
-    Log file path.
+    Set a function to allow custom filtering or sorting.
+    Below example implements removing duplicates.
 
-g:asyncomplete_auto_completeopt                       *g:asyncomplete_auto_completeopt*
+      function! s:my_asyncomplete_preprocessor(ctx, matches) abort
+        let l:dict = {}
+        let l:items = []
+        for [l:source_name, l:matches] in items(a:matches)
+          for l:item in l:matches['items']
+            if l:item['word'] =~ '^' . a:options['base']
+              if !has_key(l:dict, l:item['word'])
+                call add(l:items, l:item)
```

Comment:
Probably, `let l:dict[l:item['word']] = 1` or something is required here?

Reply:
fixed.